src/wallet3/wallet.hpp

```diff
@@ -21,6 +22,20 @@ namespace oxenmq
 
 namespace wallet
 {
+  template <typename FileType>
+  fs::path file_path_from_default_datadir(const Config& c, const FileType& filename)
+  {
+    if constexpr (std::is_same_v<FileType, std::string_view>) if (filename == ":memory:")
```

Comment:
This is quite a fragile condition: it will be entered if I call as:
```C++
file_path_from_default_datadir(conf, ":memory:"sv);
```
but not if I call it as:
```C++
file_path_from_default_datadir(conf, ":memory:");
```
or
```C++
std::string f = ":memory:";
file_path_from_default_datadir(conf, f);
```
which is rather surprising behaviour.

Reply:
Does that just change it to `(std::is_same_v<FileType, std::string_view> || std::is_same_v<FileType, std::string>)`? or is there a better type_trait(unsure if thats the correct terminology) to use here

Reply:
Could it just be a `fs::path` argument?